test/widgets/new_dm_sheet_test.dart

```diff
@@ -0,0 +1,329 @@
+
+      checkUserSelected(tester, user, false);
+      checkUserSelected(tester, eg.selfUser, false);
+      checkComposeButtonEnabled(tester, false);
+
+      await tester.tap(findUserTile(user));
+      await tester.pump();
+      checkUserSelected(tester, user, true);
+      checkComposeButtonEnabled(tester, true);
+
+      await tester.tap(findUserTile(user));
+      await tester.pump();
+      checkUserSelected(tester, user, false);
+      checkComposeButtonEnabled(tester, false);
+    });
+
+    testWidgets('other user selection deselects self user', (tester) async {
+      final otherUser = eg.user(fullName: 'Other User');
+      await setupSheet(tester, users: [eg.selfUser, otherUser]);
+
+      await tester.tap(findUserTile(eg.selfUser));
+      await tester.pump();
+      checkUserSelected(tester, eg.selfUser, true);
+      check(find.text(eg.selfUser.fullName)).findsExactly(2);
+
+      await tester.tap(findUserTile(otherUser));
+      await tester.pump();
+      checkUserSelected(tester, otherUser, true);
+      check(find.text(eg.selfUser.fullName)).findsNothing();
+    });
+
+    testWidgets('other user selection hides self user', (tester) async {
+      final otherUser = eg.user(fullName: 'Other User');
+      await setupSheet(tester, users: [eg.selfUser, otherUser]);
+
+      check(find.text(eg.selfUser.fullName)).findsOne();
+
+      await tester.tap(findUserTile(otherUser));
+      await tester.pump();
+      check(find.text(eg.selfUser.fullName)).findsNothing();
+    });
+
+    testWidgets('can select multiple users', (tester) async {
+      final user1 = eg.user(fullName: 'Test User 1');
+      final user2 = eg.user(fullName: 'Test User 2');
+      await setupSheet(tester, users: [user1, user2]);
+
+      await tester.tap(findUserTile(user1));
+      await tester.pump();
+      await tester.tap(findUserTile(user2));
+      await tester.pump();
+      checkUserSelected(tester, user1, true);
+      checkUserSelected(tester, user2, true);
+    });
+  });
+
+  group('navigation to DM Narrow', () {
+    Future<void> runAndCheck(WidgetTester tester, {
+      required List<User> users,
+      required String expectedAppBarTitle,
+    }) async {
+      await setupSheet(tester, users: users);
+
+      final context = tester.element(find.byType(NewDmPicker));
+      final store = PerAccountStoreWidget.of(context);
+      final connection = store.connection as FakeApiConnection;
+
+      connection.prepare(
+        json: eg.newestGetMessagesResult(foundOldest: true, messages: []).toJson());
+      for (final user in users) {
+        await tester.tap(findUserTile(user));
+        await tester.pump();
+      }
+      await tester.tap(findComposeButton);
+      await tester.pumpAndSettle();
+      check(find.widgetWithText(ZulipAppBar, expectedAppBarTitle)).findsOne();
+
+      check(find.byType(ComposeBox)).findsOne();
+    }
+
+    testWidgets('navigates to self DM on Next', (tester) async {
```

Comment:
This test name is out of date:
```suggestion
    testWidgets('navigates to self DM', (tester) async {
```